### PR TITLE
Add V1 file endpoints Register & Ingest

### PR DESCRIPTION
**Pull request by skairunner**
I'm unsure how to handle having new data structures that have more or less the exact same information as those from the v0 API. I also didnt add Delete because I wasn't sure if we needed that -- I assume deleting would happen via seismicstore.

### CODE REVIEW

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 1]
```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
...
     SearchSpec spec = 1;
 }
+
+message RegisterFileRequest {
+    Identifier survey = 1;                                          // [required] Either name or id of a survey previously registered with /RegisterSurvey
+    string path = 2;                                                // [required if file is not synthetic] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                                // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    map<string, string> metadata = 4;                               // [optional]
+    CRS crs = 5;                                                 // [required] Official name of the CRS used. Example: "EPSG:23031"
+    google.protobuf.BoolValue is_temporary = 6;                     // [optional] tells whether file is temporary (writeable) or not. False by default
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:236]: Remove

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 2]
```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
...
+    google.protobuf.Int32Value inline_offset = 7;                   // [optional] Inline number field in the trace headers. Defaults to 189 as per the SEG-Y rev1 specification
+    google.protobuf.Int32Value crossline_offset = 8;                // [optional] Crossline number field in the trace headers. Defaults to 193 as per the SEG-Y rev1 specification
+    google.protobuf.Int32Value cdp_x_offset = 9;                    // [optional] X coordinate of ensemble (CDP) position in trace headers. Defaults to 181 as per the SEG-Y rev1 specification
+    google.protobuf.Int32Value cdp_y_offset = 10;                   // [optional] Y coordinate of ensemble (CDP) position in trace headers. Defaults to 185 as per the SEG-Y rev1 specification
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:240]: Maybe extract to a separate message?

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 3]
```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
...
+    ExternalId external_id = 11;                                    // [optional] An external identifier - matches service contract field
+    /**
+    [optional] Multiplier for CDP-X and CDP-Y values, overrides scalar factor obtained from trace header.
+    Must be in the range (0,1].
+    Note that this is a floating point multiplier used directly to scale CDP-X and CDP-Y values, and it is not interpreted in the manner of the source group scalar trace header
+    field in the SEG-Y specification. That is: To divide by 100, specify 0.01, not -100.
+    Negative values, 0 and values greater than 1 are not permitted when registering a file, although 0 may be used in the EditFile call to unset this field.
+    **/
+    google.protobuf.FloatValue source_group_scalar_override = 12;
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:249]: Include with header offsets' message?

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 4]
```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
 message SearchFilesRequest {
     SearchSpec spec = 1;
 }
+
+message RegisterFileRequest {
+    Identifier survey = 1;                                          // [required] Either name or id of a survey previously registered with /RegisterSurvey
+    string path = 2;                                                // [required if file is not synthetic] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                                // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    map<string, string> metadata = 4;                               // [optional]
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:234]: At some point we should define som limits on the metadata object size. ie "no more than 100 keys, max 10 KB of data"

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 5]
```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
 message SearchFilesRequest {
     SearchSpec spec = 1;
 }
+
+message RegisterFileRequest {
+    Identifier survey = 1;                                          // [required] Either name or id of a survey previously registered with /RegisterSurvey
+    string path = 2;                                                // [required if file is not synthetic] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:232]: I am wondering if we should have the full cloud storage URL in this place instead. It's kinda nice to have a name though, but honestly that's the externalid... Hmm!

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto [hunk 6]
```diff
@@ -59,6 +59,15 @@ message Survey {
     CustomSurveyCoverage custom_coverage = 7; // User-provided coverage override
 }
 
+// File or dataset or cube derived from a single SEG-Y file
+message FileV1 {
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_datatypes.proto:63]: CloudStorageFile?

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto [hunk 7]
```diff
@@ -59,6 +59,15 @@ message Survey {
...
+    string uuid = 1;
+    string name = 2;
+    map<string, string> metadata = 3;
+    bool is_temporary = 4;
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_datatypes.proto:67]: remove?

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 8]
```diff
@@ -226,3 +226,81 @@ message SegYSeismicResponse {
...
+           If the last successful step was 3 (INSERT_DATA), and no start step is selected, only coverage will be
+           computed. Previously ingested headers and traces will be preserved in this case.
+        * Forcing reingestion:
+           If start_step is 1 (INSERT_FILE_HEADERS), regardless of previous ingestion status, all headers
+           and data will be deleted and the whole file will be forcefully reingested
+        * Forcing a single step to be rerun:
+           If the file is ingested (last step is COMPUTE_COVERAGE), by using start_step = 4 (COMPUTE_COVERAGE),
+           for example, only coverage will be computed while headers and traces will be preserved
+    **/
+    FileStep start_step = 2;
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:283]: Omit for now?

File: cognite/seismic/protos/v1/seismic_service_messages.proto (hunk 5 above)
Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_messages.proto:232]: What's the difference between the 'path' and the 'full cloud storage url'?

File: cognite/seismic/protos/v1/seismic_service_messages.proto (hunk 2 above)
Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_messages.proto]: 👍

File: cognite/seismic/protos/v1/seismic_service_messages.proto (hunk 5 above)
Comment (audunska) [reply cognite/seismic/protos/v1/seismic_service_messages.proto]: I guess 'full url' = 'path' / 'name'

**Comment by skairunner** [comment]
No longer required

**Pull request closed without merging**